### ISSUE TITLE
Improve error message related to analyzer crash

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/sdk#50911]
Related issues:
- https://github.com/dart-lang/sdk/issues/49281
- https://github.com/dart-lang/sdk/issues/50594

I am currently facing a crash when running `dart analyze` (see CI logs: https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709):

```
dart analyze
  shell: /usr/bin/bash -e {0}
  env:
    FLUTTER_ROOT: /opt/hostedtoolcache/flutter/stable-[3](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:3).3.10-x6[4](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:4)
    PUB_CACHE: /home/runner/.pub-cache
Analyzing dart_custom_lint...
../../runtime/vm/message_snapshot.cc: [5](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:5)7[6](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:6): error: expected: !lib.IsNull()
version=2.18.6 (stable) (Tue Dec 13 21:15:14 2022 +0000) on "linux_x64"
pid=218[7](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:8), thread=222[8](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:9), isolate_group=main(0x563b3[9](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:10)a1f800), isolate=main(0x563b39d58800)
isolate_instructions=563b3732f580, vm_instructions=563b3732f580
  pc 0x0000563b3759457c fp 0x00007f8626a7b3[10](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:11) dart::Profiler::DumpStackTrace(void*)+0x7c
  pc 0x0000563b3732f734 fp 0x00007f8626a7b3f0 dart::Assert::Fail(char const*, ...) const+0x84
  pc 0x0000563b37505a5a fp 0x00007f8626a7b450 /opt/hostedtoolcache/flutter/stable-3.3.10-x64/bin/cache/dart-sdk/bin/dart+0x20a7a5a
  pc 0x0000563b374fea04 fp 0x00007f8626a7b4b0 dart::MessageDeserializer::Deserialize()+0x244
  pc 0x0000563b374ff999 fp 0x00007f8626a7b610 dart::ReadMessage(dart::Thread*, dart::Message*)+0x159
  pc 0x0000563b374d0de8 fp 0x00007f8626a7bba0 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message> >)+0xc8
  pc 0x0000563b374fa59d fp 0x00007f8626a7bc10 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)+0x14d
  pc 0x0000563b374fac7f fp 0x00007f8626a7bc60 dart::MessageHandler::TaskCallback()+0x1df
  pc 0x0000563b3761d6b8 fp 0x00007f8626a7bce0 dart::ThreadPool::WorkerLoop(dart::ThreadPool::Worker*)+0x148
  pc 0x0000563b3761db0d fp 0x00007f8626a7bd10 dart::ThreadPool::Worker::Main(unsigned long)+0x6d
  pc 0x0000563b37590758 fp 0x00007f8626a7bdd0 /opt/hostedtoolcache/flutter/stable-3.3.10-x64/bin/cache/dart-sdk/bin/dart+0x2[13](https://github.com/invertase/dart_custom_lint/actions/runs/3849267491/jobs/6558063709#step:10:14)2758
-- End of DumpStackTrace
Bad state: The analysis server crashed unexpectedly
```

My understanding from the linked issues is that this is probably a bug caused by the SendPort changes.

But the issue with this error is that it's so cryptic, there's no way for me to move forward. With this error, I have no clue whether the problem is in my code or not, and if so what piece of code.

If this is indeed caused by not being able to send a LinkedHashMap to SendPort.send, could the error be updated to be more helpful?

Some ideas:
- Have a nice error message explaining that `LinkedHashMap` was sent
- have a stacktrace which points to the `SendPort.send` that failed.
- Have the error message include a `toString` of the failing object (to have some context about what caused this)

**Comment by mraleph** [comment]
@rrousselGit the issue AFAIK should have been fixed by @rmacnak-google in a8fe399c79eedd1e0bce072719bd142f25138f70 - instead of assertion failure you should be getting an exception now.  

Could you try on bleeding edge or dev channels?

**Comment by rrousselGit** [comment]
@mraleph For reference, my crash happened on the "stable" channel of Flutter. So afaik the SendPort changes shouldn't even be involved, right?

Surprisingly I've switched my CI to master and face no crash or errors whatsoever.
